services/pull/patch.go

```diff
@@ -29,6 +29,17 @@ import (
 	"github.com/gobwas/glob"
 )
 
+// getDiffOrPatch generates either diff or formatted patch data between given revisions
+func getDiffOrPatch(repo *git.Repository, compareString string, w io.Writer, patch, binary bool) error {
+	if patch {
+		return repo.GetPatch(compareString, w)
+	}
+	if binary {
+		return repo.GetDiffBinary(compareString, w)
+	}
+	return repo.GetDiff(compareString, w)
+}
```

Comment:
There is only one call to this function? Maybe it could be removed and just write the if/else in the caller.